### DISCUSSION TITLE
How do you guys see the host disk space?

### DISCUSSION THREAD

**Discussion by steveiliop56** [netdata/netdata#16444] [Q&A]
Hello!

I am trying to make a container display the available disk usage. I tried mounting / to /mnt/host but no luck it isn't reliable. So since you guys do the best job with monitoring how do you measure the host disk usage when running in docker?

**Comment by ilyam8** [comment]
Hey, what do you mean is not reliable? To find space/inodes stats ND uses [statfs()](https://man7.org/linux/man-pages/man2/statfs.2.html) for all mount points (`/proc/self/mountinfo`). Therefore, if the mount point is not mounted inside the container, it is impossible to get its statistics from inside the container. 

Not a specialist, but I guess you can `-v /:/host/root:ro` and you will get disk space stats for all host mounts ("Mounted on" in `df`). This should be enough if you don't add mount points in runtime - I believe by default mount propagation doesn't work. See [configure bind propagation](https://docs.docker.com/storage/bind-mounts/#configure-bind-propagation), I think `rslave` should be ok - if the host mounts anything inside the volume mount, the container will see it mounted too.

**Reply by steveiliop56** [reply]
Hello!

What I use right now is this ```const disks = await si.fsSize();
    const disk0 = disks.find((disk) => disk.mount.startsWith('/mnt/host'));```

But that doesn't work at all on ubuntu, works on debian, doesn't work on macos. So if I understood correctly I can find disk usage and stuff by only using /proc/self/mountinfo? Also could you also give the link to which file is responsible for disk usage in netdata so I can check it out?
